lib/aws-xray-sdk/plugins/ecs.rb

```diff
@@ -10,13 +10,80 @@ module ECS
 
       ORIGIN = 'AWS::ECS::Container'.freeze
 
+      # Only compatible with v4!
+      # The v3 metadata url does not contain cloudwatch informations
+      METADATA_ENV_KEY = 'ECS_CONTAINER_METADATA_URI_V4'
```

Comment:
Here we have a strong dependency with Fargate 1.4, otherwise this environment variable will not be defined:
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4-fargate.html

Sadly the necessary information are not exposed with the meta-information of the v3:
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v3-fargate.html